esp-hal/src/lib.rs

```diff
@@ -738,24 +742,58 @@ use crate::{
 pub struct Config {
```

Comment:
FWIW, I think this is okay to live in the `config` module _but_ we should reexport it from lib.rs for convenience.

Reply:
But then we should probably not make it public in the `config` module, the same struct appearing in two places will be confusing.